### PR TITLE
Add generation for getting objects from requests

### PR DESCRIPTION
**Pull request by thedadams**
In order to validate objects, the new and old objects, if they exist,
should be pull from the webhook request. This process is identical for
all objects except that the return types would be different. This is a
candidate for code generation, and this generation is added.

### CODE REVIEW

File: pkg/codegen/template.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+	oldObject, err := request.DecodeOldObject()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return oldObject.({{ .Type }}), object.({{ .Type }}), nil
+}
+
+func {{ .Name }}ObjectFromRequest(request *webhook.Request) ({{ .Type }}, error) {
```
Comment (dramich) [inline pkg/codegen/template.go:40]:
```suggestion
func {{ .Name }}FromRequest(request *webhook.Request) ({{ .Type }}, error) {
```

File: pkg/codegen/template.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	{{ range .types }}
+	"{{ .Package }}"{{ end }}
+	"github.com/rancher/wrangler/pkg/webhook"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+{{ range .types }}
+
+func {{ .Name }}OldAndNewFromRequest(request *webhook.Request) ({{ .Type }}, {{ .Type }}, error) {
```
Comment (dramich) [inline pkg/codegen/template.go:16]: Add a comment above each function to describe what it does

File: main.go [hunk 3]
```diff
@@ -1,5 +1,5 @@
 //go:generate go run pkg/codegen/cleanup/main.go
-//go:generate go run pkg/codegen/main.go
+//go:generate go run pkg/codegen/main.go pkg/codegen/template.go
```
Comment (dramich) [inline main.go:2]: Is this change needed? that file is part of the main package and doesn't run any code

Comment (thedadams) [reply main.go:2]: Yes, it is required: `go run` doesn't look at files not passed as arguments when compiling. Without this change, the template isn't found because it is in `pkg/codegen/template.go`.

File: pkg/codegen/template.go (hunk 2 above)
Comment (thedadams) [reply pkg/codegen/template.go:18]: Done.

File: pkg/codegen/template.go (hunk 1 above)
Comment (thedadams) [reply pkg/codegen/template.go]: Done.

**Review by ryansann — APPROVED** [review]
LGTM, nice cleanup in the handlers

**Review by dramich — APPROVED** [review]
